src/api/worker/rest/DefaultEntityRestCache.ts

```diff
@@ -261,7 +261,8 @@ export class DefaultEntityRestCache implements EntityRestCache {
 		return this.entityRestClient.setupMultiple(listId, instances)
 	}
 
-	update<T extends SomeEntity>(instance: T): Promise<void> {
+	async update<T extends SomeEntity>(instance: T): Promise<void> {
+		await this.storage.put(instance)
```

Comment:
I think this might be wrong if the request fails for any reason, maybe it would be more correct to insert it after the request is done?

Not sure how to test this - maybe setting something up to set the read/unread flag on a nonexistent mail

Reply:
That is a good point, I'll switch to updating the cache after the request is finished as per your suggestion.

Reply:
it's really a both-or-nothing situation, but the put should fail less often and has the entity event that comes later as a fallback, so we could try - catch - ignore errors there.

I'm also a bit concerned that this would be the first instance where we trigger caches not from the entity events but preempt it, but can't think of a reason not to do it right now

Reply:
> I'm also a bit concerned that this would be the first instance where we trigger caches not from the entity events but preempt it, but can't think of a reason not to do it right now

Fair enough to be concerned, this is the reason the test notes are so broad. 
Can you think of an example where the entity events trigger the cache? To my knowledge `entityEventsReceived()` in `CalendarModel` doesn't trigger the cache directly.

Reply:
the main way the cache is kept up-to-date is `DefaultEntityRestCache.entityEventsReceived()` . Afaik it's called right before any other entity listeners.

Reply:
`DefaultEntityRestCache.entityEventsReceived()` was not in the call stack when I debugged the issue, I'll take another look and see if I can add a call to it.

Reply:
yes, the update call completely bypasses the cache (it's semantically read-only for most of the client code).

once the update has been processed by the server, an entity update batch is generated and sent out to all connected clients and stored for 40 days for clients that are offline, and that event batch is what triggers the 
update of the cache, it loads updated entities that are already cached and directly stores them. this can be weeks later. 

this storage.put should be OK because we know the update succeeded and we will get an event update soon anyway

that reminds me, we have a few uncached entity types and we need to make sure they stay uncached.

Reply:
I feel  pretty strongly that we should not do it, at least for now.

This can introduce all kinds of fun inconsistencies. Entity events should be applied in the correct order, otherwise fun things happen.

Commit also doesn't explain how this helps the issue.

Reply:
> I feel pretty strongly that we should not do it, at least for now.
> 
> This can introduce all kinds of fun inconsistencies. Entity events should be applied in the correct order, otherwise fun things happen.
> 
> Commit also doesn't explain how this helps the issue.

We noticed that the event name has not being updated at the time the alarm is scheduled. The CalendarEvent is updated before the alarm, but when the alarm searches for the Event info, it receives a cached version of it with incorrect information about the event, the name is one of these incorrect information.

Reply:
I thought we re-create the event if the summary is changed?

We have a code in place to resolve this issue already:

https://github.com/tutao/tutanota/blob/43acfdc6ef806af5d473f83cb110b882063a1342/src/calendar/model/CalendarModel.ts#L608-L654

but it's for `CREATE` event (which I thought should be the case for changing summary). We should check why it's not the case